fastestimator/trace/io/model_eval.py

```diff
@@ -0,0 +1,451 @@
+            self.json_summary["tests"].append(case_dict)
+
+        for case in self.aggregate_cases:
+            case_dict = {"test_type": "aggregate", "description": case.description}
+            case_dict["passed"] = self._to_serializable(case.result)
+            case_dict["inputs"] = case.input_val
+            self.json_summary["tests"].append(case_dict)
+
+        self.json_summary["execution_time(s)"] = time() - self.json_summary["execution_time(s)"]
+
+        self._dump_json()
+        self._init_document()
+        self._document_test_result()
+        self._dump_pdf()
+
+    def _initialize_json_summary(self) -> None:
+        """Initialize json summary
+        """
+        self.json_summary = {
+            "title": self.test_title, "timestamp": str(datetime.now()), "execution_time(s)": time(), "tests": []
+        }
+
+    def _sanitize_report_name(self) -> None:
+        """Sanitize report name and make it class attribute
+        """
+        exp_name = self.system.summary.name
+        if not exp_name:
+            raise RuntimeError("TestReport require an experiment name to be provided in estimator.fit()")
+        # Convert the experiment name to a report name (useful for saving multiple experiments into same directory)
+        report_name = "".join('_' if c == ' ' else c for c in exp_name
+                              if c.isalnum() or c in (' ', '_')).rstrip("_").lower()
+        report_name = re.sub('_{2,}', '_', report_name) + "_ModelEval"
+        self.report_name = report_name or '_ModelEval'
+
+    def _init_document(self) -> None:
+        """Initialize latex document
+        """
+        self.doc = Document(geometry_options=['lmargin=2cm', 'rmargin=2cm', 'tmargin=2cm', 'bmargin=2cm'])
+        self.doc.packages.append(Package(name='placeins', options=['section']))
+        self.doc.packages.append(Package(name='float'))
+
+        self.doc.preamble.append(NoEscape(r'\aboverulesep=0ex'))
+        self.doc.preamble.append(NoEscape(r'\belowrulesep=0ex'))
+        self.doc.preamble.append(NoEscape(r'\renewcommand{\arraystretch}{1.2}'))
+
+        self.doc.preamble.append(Command('title', self.json_summary["title"]))
+        self.doc.preamble.append(Command('author', f"FastEstimator {fe.__version__}"))
+        self.doc.preamble.append(Command('date', NoEscape(r'\today')))
+        self.doc.append(NoEscape(r'\maketitle'))
+
+        # new column type for tabularx
+        self.doc.preamble.append(NoEscape(r'\newcolumntype{Y}{>{\centering\arraybackslash}X}'))
+
+        # add seqinsert hyphentation
+        self.doc.preamble.append(NoEscape(r'\def\seqinsert{\-}'))
+
+        # TOC
+        self.doc.append(NoEscape(r'\tableofcontents'))
+        self.doc.append(NoEscape(r'\newpage'))
+
+    def _document_test_result(self) -> None:
+        """Document test results including test summary, passed tests, and failed tests.
+        """
+        self.test_id = 1
```

Comment:
this probably shouldn't be a member variable, but if it is it should be defined in init and then just re-initialized here

Reply:
The test ID will be needed in four places. I think it is okay to make it a member variable based on the fact that self.doc is also a member variable.